### ISSUE TITLE
Code generated using code snippets does not match parentheses correctly inside

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#68876]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Code-generated-using-code-snippets-does-/10405378)._

---
1. input cw
2. press tab 键
3. input typeof()

expect: Console.WriteLine(typeof())
actual: Console.WriteLine(typeof()

---
#### Feedback Bot on 7/3/2023, 00:59 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
This is interesting in that we likely need to get the editor to enter into 'brace matching mode'.  despite the user *not* having typed the `(` for the `WriteLine` themselves.  Seems like it could be hard to make work.  But the user scenario is very real here.

**Comment by Autumn-one** [comment]
> This is interesting in that we likely need to get the editor to enter into 'brace matching mode'. despite the user _not_ having typed the `(` for the `WriteLine` themselves. Seems like it could be hard to make work. But the user scenario is very real here.

Yes, the biggest problem here is that this operation occurs very frequently, and I rely heavily on code snippets when writing the code, which causes me to manually fix the parentheses every time.
